src/mongo_types.ts

```diff
@@ -541,3 +525,60 @@ export type NestedPathsOfType<TSchema, Type> = KeysOfAType<
   },
   Type
 >;
+
+/**
+ * @public
+ * @experimental
+ */
+export type DotNotationFilter<TSchema> =
```

Comment:
Since the expectation would now be that folks can use this annotation directly (instead of via the find call) Can we add more to the docs here? I'm thinking we can copy some of the language from the migration guide + 4.3.0 release notes w/ a code snippet

Reply:
My thoughts are that the existing documentation and release notes are enough, and that we don't want to provide documentation on an experimental (read: unsupported) feature.

Reply:
We're still going to have [this section](https://www.mongodb.com/docs/drivers/node/current/fundamentals/typescript/#type-safety-and-dot-notation) on the docs page (updated for this change) so we can link to that. In terms of code snippet, it is new for us to have a "feature" that's opt-in TS, I was thinking something small that just demonstrates it being used as an annotation.

We should get clarity on messaging here, we have other features marked experimental that are supported so if unsupported is the plan here we might want a different marker

Reply:
Are we going to have that docs page?  I assumed that would be removed for 5.x

Reply:
If that's the case can you clarify that we want it removed in the "Documentation Changes Summary", I thought it was remaining because it says we're keeping the type (which is true) so I thought the docs stayed too

From slack, `@unsupported - xxx` would be a preferred doc tag where xxx is some language that describes what that means